diff -Naurb kfabric.orig/tests/cxi/single_client/test_rx_cancel.c kfabric/tests/cxi/single_client/test_rx_cancel.c
--- kfabric.orig/tests/cxi/single_client/test_rx_cancel.c	2025-11-10 13:41:17
+++ kfabric/tests/cxi/single_client/test_rx_cancel.c	2025-11-10 13:41:45
@@ -17,6 +17,7 @@
 #include <linux/kthread.h>
 #include <linux/random.h>
 #include <linux/mm.h>
+#include <linux/vmalloc.h>
 
 #include "../../../prov/cxi/kcxi_prov.h"
 
