diff --git a/knl/core/base/sl_core_timer_link.c b/knl/core/base/sl_core_timer_link.c
index 9fa93a4..8119eb6 100644
--- a/knl/core/base/sl_core_timer_link.c
+++ b/knl/core/base/sl_core_timer_link.c
@@ -81,7 +81,7 @@ void sl_core_timer_link_timeout(struct timer_list *timer)
 	struct sl_core_timer_link_info *info;
 	struct sl_core_link            *core_link;
 
-	info = from_timer(info, timer, timer);
+	info = timer_container_of(info, timer, timer);
 	core_link = info->data.link;
 
 	sl_core_log_dbg(core_link, LOG_NAME,
@@ -97,5 +97,5 @@ void sl_core_timer_link_end(struct sl_core_link *core_link, u32 timer_num)
 		"end %s (core_link = 0x%p, timer_num = %u)",
 		core_link->timers[timer_num].data.log, core_link, timer_num);
 
-	del_timer_sync(&(core_link->timers[timer_num].timer));
+	timer_delete_sync(&(core_link->timers[timer_num].timer));
 }
diff --git a/knl/core/base/sl_core_timer_llr.c b/knl/core/base/sl_core_timer_llr.c
index b9298cb..a095a2f 100644
--- a/knl/core/base/sl_core_timer_llr.c
+++ b/knl/core/base/sl_core_timer_llr.c
@@ -64,7 +64,7 @@ void sl_core_timer_llr_timeout(struct timer_list *timer)
 	struct sl_core_timer_llr_info *info;
 	struct sl_core_llr            *core_llr;
 
-	info = from_timer(info, timer, timer);
+	info = timer_container_of(info, timer, timer);
 	core_llr = info->data.core_llr;
 
 	sl_core_log_dbg(core_llr, LOG_NAME,
@@ -81,5 +81,5 @@ void sl_core_timer_llr_end(struct sl_core_llr *core_llr, u32 timer_num)
 		"end %s (core_llr = 0x%p, timer_num = %u)",
 		core_llr->timers[timer_num].data.log, core_llr, timer_num);
 
-	del_timer_sync(&(core_llr->timers[timer_num].timer));
+	timer_delete_sync(&(core_llr->timers[timer_num].timer));
 }
diff --git a/knl/ctl/sl_ctl_link_fec_priv.c b/knl/ctl/sl_ctl_link_fec_priv.c
index b2685b4..c0970e1 100644
--- a/knl/ctl/sl_ctl_link_fec_priv.c
+++ b/knl/ctl/sl_ctl_link_fec_priv.c
@@ -489,7 +489,7 @@ void sl_ctl_link_fec_mon_timer(struct timer_list *timer)
 {
 	struct sl_ctl_link *ctl_link;
 
-	ctl_link = from_timer(ctl_link, timer, fec_mon_timer);
+	ctl_link = timer_container_of(ctl_link, timer, fec_mon_timer);
 
 	sl_ctl_log_dbg(ctl_link, LOG_NAME, "monitor timer");
 
@@ -513,7 +513,7 @@ void sl_ctl_link_fec_mon_stop(struct sl_ctl_link *ctl_link)
 		return;
 	}
 
-	del_timer_sync(&ctl_link->fec_mon_timer);
+	timer_delete_sync(&ctl_link->fec_mon_timer);
 }
 
 #define SL_CTL_LINK_FEC_LIMIT_25   25781250000ULL
